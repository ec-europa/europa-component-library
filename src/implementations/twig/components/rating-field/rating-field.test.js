import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe } from 'vitest-axe';
import specs from '@ecl/specs-component-rating-field/demo/data';

const data = specs.input;

describe('Rating field', () => {
  const template = '@ecl/rating-field/rating-field.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, data, true)),
      ).toHaveNoViolations();
    });
  });
});
