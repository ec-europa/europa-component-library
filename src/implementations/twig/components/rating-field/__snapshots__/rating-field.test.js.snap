// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating field Default renders correctly 1`] = `
<jest>
  <fieldset
    aria-describedby="helper-id-1"
    class="ecl-form-group"
  >
    <legend
      class="ecl-form-label"
    >
      Please Rate
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="helper-id-1"
    >
      This is the group's helper text.
    </div>
    <div
      class="ecl-rating-field"
    >
      <input
        class="ecl-rating-field__input"
        id="rating-5"
        name="rating"
        required=""
        type="radio"
        value="5"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-5"
        title="5 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-4"
        name="rating"
        required=""
        type="radio"
        value="4"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-4"
        title="4 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-3"
        name="rating"
        required=""
        type="radio"
        value="3"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-3"
        title="3 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-2"
        name="rating"
        required=""
        type="radio"
        value="2"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-2"
        title="2 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-1"
        name="rating"
        required=""
        type="radio"
        value="1"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-1"
        title="1 star"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
    </div>
  </fieldset>
</jest>
`;

exports[`Rating field Default renders correctly with extra attributes 1`] = `
<jest>
  <fieldset
    aria-describedby="helper-id-1"
    class="ecl-form-group"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <legend
      class="ecl-form-label"
    >
      Please Rate
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="helper-id-1"
    >
      This is the group's helper text.
    </div>
    <div
      class="ecl-rating-field"
    >
      <input
        class="ecl-rating-field__input"
        id="rating-5"
        name="rating"
        required=""
        type="radio"
        value="5"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-5"
        title="5 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-4"
        name="rating"
        required=""
        type="radio"
        value="4"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-4"
        title="4 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-3"
        name="rating"
        required=""
        type="radio"
        value="3"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-3"
        title="3 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-2"
        name="rating"
        required=""
        type="radio"
        value="2"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-2"
        title="2 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-1"
        name="rating"
        required=""
        type="radio"
        value="1"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-1"
        title="1 star"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
    </div>
  </fieldset>
</jest>
`;

exports[`Rating field Default renders correctly with extra class names 1`] = `
<jest>
  <fieldset
    aria-describedby="helper-id-1"
    class="ecl-form-group custom-class custom-class--test"
  >
    <legend
      class="ecl-form-label"
    >
      Please Rate
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="helper-id-1"
    >
      This is the group's helper text.
    </div>
    <div
      class="ecl-rating-field"
    >
      <input
        class="ecl-rating-field__input"
        id="rating-5"
        name="rating"
        required=""
        type="radio"
        value="5"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-5"
        title="5 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-4"
        name="rating"
        required=""
        type="radio"
        value="4"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-4"
        title="4 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-3"
        name="rating"
        required=""
        type="radio"
        value="3"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-3"
        title="3 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-2"
        name="rating"
        required=""
        type="radio"
        value="2"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-2"
        title="2 stars"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
      <input
        class="ecl-rating-field__input"
        id="rating-1"
        name="rating"
        required=""
        type="radio"
        value="1"
      />
      <label
        class="ecl-rating-field__label"
        for="rating-1"
        title="1 star"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-filled"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-filled"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-rating-field__icon-outline"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#star-outline"
          />
        </svg>
      </label>
    </div>
  </fieldset>
</jest>
`;
